server/middleware/validation.js

```diff
@@ -29,6 +32,24 @@ const validators = {
         }
         return null;
     },
+    isValidDay({ label, value, message }) {
+        if (getDay(value) > new Date(getYear(value), getMonth(value), 0).getDate() || getDay(value) < 1) {
+            return message || validationErrors.isValidDay(label);
+        }
+        return null;
+    },
+    isValidMonth({ label, value, message }) {
+        if (getMonth(value) < 1 || getMonth(value) > 12) {
+            return message || validationErrors.isValidMonth(label);
+        }
+        return null;
+    },
+    isValidYear({ label, value, message }) {
```

Comment:
Not sure about this. 1700 _is_ a year, after all. It's just not a year we'd expect to see in correspondence metadata. This might break some edge-case date of birth fields: 100 years is still within a human lifespan.

Use a more precise name for this function. This isn't an `isValidYear()` -- it's an "is now±100y"
Consider using more precise message too: "Year must be after 1911".

Reply:
Agreed and updated